core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -299,33 +299,41 @@ protected void doPost(HttpServletRequest httpServletRequest,
 			        }
 	            }
             }
-
-            // Dispatch to query result page
-            if (action != null && action.equals(ACTION_SUBMIT) && (!errorsExist)) {
-                CohortDetails cohortDetails;
-                if (httpServletRequest.getParameter(CANCER_STUDY_ID).equals("all") &&
-                    httpServletRequest.getParameter(CANCER_STUDY_LIST) != null) { // multiple studies
+            
+            if(httpServletRequest.getAttribute(CANCER_STUDY_ID) != null) {
+            		CohortDetails cohortDetails;
+                if (httpServletRequest.getParameter(CANCER_STUDY_ID).equals("all")) { // multiple studies
```

Comment:
@kalletlak maybe ok not to catch `httpServletRequest.getParameter(CANCER_STUDY_LIST) != null`, but would be better to catch that as error for malformatted request. Instead of getting an exception when doing the `split` further down